test/fixtures/stdout-producer.js

```diff
@@ -0,0 +1,13 @@
+'use strict';
+
+const add = 'test\n';
+
+var str = '';
+for (var i = 0; i < 100000; i++) {
+  str += add;
+}
+
+str += 'hey\n';
+
+process.stdout.write(str);
+process.exit();
```

Comment:
`process.exit()` unconditionally terminates the process.  Even when stdio is synchronous, there is no guarantee that the other end will have received it all.